packages/curve-ui-kit/src/themes/tabs/mui-tabs.ts

```diff
@@ -13,51 +18,134 @@ export const TABS_HEIGHT_CLASSES = { small, medium, large }
+ *    - Handles varying border heights and paddings cleanly
+ *
+ * 3. Simpler state management:
+ *    - No need for transparent border fallbacks
+ *    - Works seamlessly for both underline/overline variants
+ *
+ * 4. The ::after element enables separate transitions for opacity and background-color:
+ *    - Hover state can smoothly transition from transparent to outline color via opacity
+ *    - Active state can instantly switch outline colors without transition
+ *    - This solves the limitation where a single CSS property (like border-color)
+ *      cannot have different transition behaviors for hover vs active states
+ *    In other words, this allows you to transition the border on and off state while
+ *    keeping color changes instantenously.
+ */
+
+export const defineMuiTab = ({ Tabs: { Transition } }: DesignSystem): Components['MuiTab'] => ({
   styleOverrides: {
     root: {
+      transition: Transition,
       textTransform: 'uppercase',
-      borderTop: '2px solid transparent',
-      minHeight: 0,
-      boxSizing: 'border-box',
+      position: 'relative',
+      boxSizing: 'content-box',
+      opacity: 1,
+      minHeight: '2rem', // Not responsive, but Sizing.md is ugly in mobile
+      minWidth: 0, // It's 90px by default in Mui, but we want it smaller
+      '&::after': {
+        content: '""',
+        position: 'absolute',
+        height: BORDER_SIZE,
+        opacity: 0,
+        transition: `opacity ${TransitionFunction}`,
+      },
     },
   },
 })
 
 type TabStyle = { Label?: string; Fill?: string; Outline?: string }
-type TabVariant = { Default: TabStyle; Hover: TabStyle; Current: TabStyle }
+type TabVariant = { Inset?: string; Default: TabStyle; Hover: TabStyle; Current: TabStyle }
 
 const tabStyle = ({ Label, Fill, Outline }: TabStyle) => ({
   color: Label,
   backgroundColor: Fill,
-  borderColor: Outline ?? 'transparent',
+  '::after': {
+    backgroundColor: Outline ?? 'transparant',
+  },
 })
 
-const tabVariant = ({ Current, Default, Hover }: TabVariant) => ({
+const tabVariant = ({ Current, Default, Hover, Inset }: TabVariant) => ({
   ...tabStyle(Default),
   '&:hover': tabStyle(Hover),
+  '&:hover::after': { opacity: 1 },
   '&.Mui-selected': tabStyle(Current),
+  '&.Mui-selected::after': { opacity: 1 },
+  '::after': Inset && {
+    inset: Inset,
+  },
 })
 
+const tabPadding = (
+  blockStart: keyof typeof Spacing,
+  blockEnd: keyof typeof Spacing,
+  inlineStart: keyof typeof Spacing,
+  inlineEnd: keyof typeof Spacing,
+) =>
+  handleBreakpoints({
+    paddingBlockStart: Spacing[blockStart],
+    paddingBlockEnd: Spacing[blockEnd],
+    paddingInlineStart: Spacing[inlineStart],
+    paddingInlineEnd: Spacing[inlineEnd],
+  })
+
+// Overlined and Underlined have common paddings and sizes.
+const tabSizesNonContained = {
+  [`&.${medium} .MuiTab-root`]: tabPadding('md', 'xs', 'md', 'md'),
+  [`&.${large} .MuiTab-root`]: tabPadding('md', 'xs', 'md', 'md'),
```

Comment:
These calls are not readable without checking the method definition.
Also, the order of the args doesn't follow the css standard (top, right, bottom, left).

```suggestion
  [`&.${medium} .MuiTab-root`]: handleBreakpoints({ padding: 'md', paddingBlockEnd: 'xs' }),
  [`&.${large} .MuiTab-root`]: handleBreakpoints({ padding: 'md', paddingBlockEnd: 'xs' }),
```

Or...
```suggestion
  [`&.${medium} .MuiTab-root`]: tabPadding({ blockStart: 'md', blockEnd: 'xs', inlineStart: 'md', inlineEnd: 'md' }),
  [`&.${large} .MuiTab-root`]: tabPadding({ blockStart: 'md', blockEnd: 'xs', inlineStart: 'md', inlineEnd: 'md'),
```

But yeah they are quite simple so not a big deal.